### ISSUE TITLE
Generate NXDomain from Lua Records

### ISSUE THREAD

**Issue by sbakhos** [PowerDNS/pdns#8551]
- Program: Authoritative
 - Issue type: Feature request

### Short description
I would like to be able to fake a NXdomain result when a lua record resolves to specific netmasks, either using the builtin view() or the if() functions. This would allow the possibility to hide records from some netmasks. 

### Usecase
To return a nxdomain error when we do not want an IP to see that a record exists. 

### Description
I am trying to hide some internal records from the public zone, without setting up a second set of servers just for the internal records. It could be useful to have something like this as a LUA record:
view({'192.168.0.0/16', '192.168.20.20'}, {'0.0.0.0/0', nxdomain()})

**Comment by Habbie** [comment]
Can you try (I'm unsure this is easy today) generating CNAMEs instead? Then you can generate a CNAME to a name that does not exist.

**Comment by rgacogne** [comment]
I don't know anything about your exact setup but please note that one needs to be very careful when mixing positive and negative answers based on the source, as it might be cached for everyone, for example by a recursor.

**Comment by Habbie** [comment]
Indeed, https://tools.ietf.org/html/rfc7871#section-7.4 says "It is RECOMMENDED that no specific behavior regarding negative answers be relied upon"

**Comment by sbakhos** [comment]
The setup I'm trying to replace is a bind setup that uses views and files.
The new setup that I have is as follows:

dnsdist as the main listener
if someone is coming from an external network, forward to auth server with useClientSubnet
if someone is coming from an internal network, forward to the recursor, which is configured to forward to the auth server for the zones we serve, also using useClientSubnet.
In all cases, setECSOverride is enabled

In the auth server, I'm trying to have views for a select few records to hide them from the internet at large. 

I will try the CNAME trick.

**Comment by sbakhos** [comment]
The CNAME doens't work very well. It does return a NXDOMAIN for the missing CNAME domain. but you can still see that the original domain exists with dig, as it is a CNAME towards something else.

**Comment by Habbie** [comment]
> The CNAME doens't work very well. It does return a NXDOMAIN for the missing CNAME domain. but you can still see that the original domain exists with dig, as it is a CNAME towards something else.

Yes, that is the behaviour I hoped for.
